In [2]:
import pandas as pd

In [3]:
outlets = pd.read_csv("Data/QSR_outlet_dataset_cleaned.csv", keep_default_na= False)
socioeconomic = pd.read_csv("Data/QSR_socioeconomic_dataset_cleaned.csv")

In [4]:
print(f"QSR outlet row and column: {outlets.shape}")
print(f"Socio-Economic row and column: {socioeconomic.shape}")

QSR outlet row and column: (2822, 17)
Socio-Economic row and column: (20, 9)


In [5]:
outlets.head()

,Outlet ID,Brand,Region,Province,Province_Shorthand,City,Menu_Category,Price_Range,Customer_Rating,Review_Count,Latitude,Longitude,Location_Type,Address,Source,Price_Range_Cleaned,Price_Range_Imputed
0,1,McDonald's,Valle d'Aosta,Aosta,AO,Aosta,Burgers,EUR 1-10,3.8,3974,45.73953,7.345432,Commercial Area,"Aosta, 11100 Aosta, Aosta Valley",https://www.mcdonalds.it/elenco-ristoranti,EUR 1-10,False
1,2,McDonald's,Liguria,Genova,GE,Genova,Burgers,EUR 1-10,3.7,2244,44.40684,8.936673,City Center,Via Venti Settembre 1,https://www.mcdonalds.it/elenco-ristoranti,EUR 1-10,False
2,3,McDonald's,Liguria,Genova,GE,Genova,Burgers,EUR 1-10,3.9,2170,44.41372,8.879941,Shopping Mall,"Via Paolo Mantovani, 17, 16149 Genova GE",https://www.mcdonalds.it/elenco-ristoranti,EUR 1-10,False
3,4,McDonald's,Liguria,Genova,GE,Genova,Burgers,EUR 1-10,2.9,303,44.41724,8.921427,Transport Hub,"Piazza Acquaverde, 16134 Genova GE",https://www.mcdonalds.it/elenco-ristoranti,EUR 1-10,False
4,5,McDonald's,Liguria,Genova,GE,Genova,Burgers,EUR 1-10,3.8,3290,44.41141,8.928781,Transport Hub,"Via di Sottoripa, 127R/133, 16124 Genova GE",https://www.mcdonalds.it/elenco-ristoranti,EUR 1-10,False


In [6]:
socioeconomic.head()

,Region,Province_Count,Region_Area,Population,Population Density,Disposable Income per capita,Total Tourism Flow,Foreign Tourism Flow,Median Income
0,Lombardia,12,23863.65,10065694,421.8,27243,41795073.0,26300388.0,32485
1,Valle d'Aosta,1,3260.90,122554,37.6,25152,3692878.0,1439669.0,32900
2,Piemonte,8,25387.07,4255006,167.6,24416,14410448.0,7180603.0,30267
3,Trentino-Alto Adige,2,13605.50,1090818,80.2,28130,55227131.0,33544804.0,37563
4,Veneto,7,18407.42,4857460,263.9,24103,71896863.0,49799037.0,32272


In [7]:
#Checking Data Types

outlets.info()

<class 'pandas.DataFrame'>
RangeIndex: 2822 entries, 0 to 2821
Data columns (total 17 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Outlet ID            2822 non-null   int64  
 1   Brand                2822 non-null   str    
 2   Region               2822 non-null   str    
 3   Province             2822 non-null   str    
 4   Province_Shorthand   2822 non-null   str    
 5   City                 2822 non-null   str    
 6   Menu_Category        2822 non-null   str    
 7   Price_Range          2822 non-null   str    
 8   Customer_Rating      2822 non-null   float64
 9   Review_Count         2822 non-null   int64  
 10  Latitude             2822 non-null   float64
 11  Longitude            2822 non-null   float64
 12  Location_Type        2822 non-null   str    
 13  Address              2822 non-null   str    
 14  Source               2822 non-null   str    
 15  Price_Range_Cleaned  2822 non-null   str    
 16 

In [8]:
#Checking Data Types

socioeconomic.info()

<class 'pandas.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 9 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Region                        20 non-null     str    
 1   Province_Count                20 non-null     int64  
 2   Region_Area                   20 non-null     float64
 3   Population                    20 non-null     int64  
 4   Population Density            20 non-null     float64
 5   Disposable Income per capita  20 non-null     int64  
 6   Total Tourism Flow            20 non-null     float64
 7   Foreign Tourism Flow          20 non-null     float64
 8   Median Income                 20 non-null     int64  
dtypes: float64(4), int64(4), str(1)
memory usage: 1.5 KB


In [9]:
# Checking missing vlaues for QSR outlet

(outlets == "").sum()

Outlet ID               0
Brand                   0
Region                  0
Province                0
Province_Shorthand      0
City                    0
Menu_Category           0
Price_Range            63
Customer_Rating         0
Review_Count            0
Latitude                0
Longitude               0
Location_Type           0
Address                 0
Source                  0
Price_Range_Cleaned     0
Price_Range_Imputed     0
dtype: int64

In [10]:
#Checking for missing values

socioeconomic.isna().sum()

Region                          0
Province_Count                  0
Region_Area                     0
Population                      0
Population Density              0
Disposable Income per capita    0
Total Tourism Flow              0
Foreign Tourism Flow            0
Median Income                   0
dtype: int64

In [11]:
# Summary statistics for the numeric variables of the QSR outlet dataset
outlets.describe()

,Outlet ID,Customer_Rating,Review_Count,Latitude,Longitude
count,2822.000000,2822.000000,2822.000000,2822.000000,2822.000000
mean,1411.500000,3.870872,1858.235294,43.795746,11.153791
std,814.785555,0.407405,2966.700272,2.165241,2.230963
min,1.000000,1.000000,1.000000,36.901520,6.892710
25%,706.250000,3.700000,248.000000,41.915950,9.204367
50%,1411.500000,3.900000,1038.500000,44.796410,11.017920
75%,2116.750000,4.100000,2636.250000,45.491147,12.543187
max,2822.000000,5.000000,59200.000000,46.679270,18.197400


From this data we can know:
    <br />1, The average customer rating is 3.870872 out of 5 for all QSRs in Italy.
    <br>2, The average outlet has about 1858 reviews. But the review counts are very spread out. So, the median would be a better measure of a typical outlet
    <br>3, The coordinates are inside Italy. As Italy has Latitude between (35.5, 47.1) and Longitude between (6.6, 18.6)

In [12]:
# Summary statistics for numeric socioeconomic variables

socioeconomic.describe()

,Province_Count,Region_Area,Population,Population Density,Disposable Income per capita,Total Tourism Flow,Foreign Tourism Flow,Median Income
count,20.000000,20.000000,2.000000e+01,20.000000,20.000000,2.000000e+01,2.000000e+01,20.000000
mean,5.350000,15103.642000,2.947141e+06,177.885000,21767.100000,2.235850e+07,1.170911e+07,29163.300000
std,3.013566,7448.375038,2.509338e+06,112.313177,3843.639132,2.019318e+07,1.369814e+07,4658.277297
min,1.000000,3260.900000,1.225540e+05,37.600000,16173.000000,4.947860e+05,4.749700e+04,22225.000000
25%,3.500000,9167.117500,1.167826e+06,95.425000,17993.000000,7.776650e+06,1.668786e+06,24696.250000
50%,5.000000,14446.425000,1.691030e+06,158.300000,22732.000000,1.524733e+07,6.999274e+06,29490.500000
75%,7.250000,22586.345000,4.551555e+06,215.525000,24583.250000,3.983087e+07,1.463992e+07,32992.000000
max,12.000000,25832.390000,1.006569e+07,421.800000,28130.000000,7.189686e+07,4.979904e+07,37563.000000


From this result we can understand:
<br>1, How large Italian regions are in population. Italian regions are very different in population size. The average region has 2.95 million people but the median has only 1.69 million population size. Meaning, very few large regions are pulling the average up. Therefore, Most Italian regions are relatively small or medium-sized, while a few large regions contain a very high share of the national population.

2, How different are regions is population density. they vary very much in how crowded they are. some regions have low density, around 37.6 people per km2 while others have 442 people per km2. Therefore, QSR outlets may be more concentrated in dense regions because dense areas have more foot traffic, more urban centers, and more commercial activity.

3, There is also a clear income difference among the regions. Higher-income regions may support more QSR outlets, especially brands with higher price ranges, but this needs to be tested later with correlation and regression.

4, The maximum tourism flow is huge compared to the minimum. Some regions receive less than 500,000 tourist flows, while the highest region receives about 71.9 million. Telling us that, tourism is concentrated in small number of regions

5, Median income also varies strongly across regions. Some regions have much stronger household income conditions than others, which may influence restaurant demand, brand location strategy, and price-range distribution.

In [13]:
# Count outlets by QSR Brand

outlets["Brand"].value_counts()


Brand
McDonald's              790
La Piadineria           528
Autogrill               282
Old Wild West           243
Alice Pizza             181
Burger King             177
Roadhouse Restaurant    168
KFC                     153
Chef Express             99
Billy Tacos              69
Starbucks                48
Spontini                 25
Popeyes                  15
Spizzico                 15
Five Guys                10
Burgez                   10
Subway                    9
Name: count, dtype: int64

In [14]:
# The percentage share of outlets by Brand
outlets["Brand"].value_counts(normalize= True) * 100

Brand
McDonald's              27.994330
La Piadineria           18.710135
Autogrill                9.992913
Old Wild West            8.610914
Alice Pizza              6.413891
Burger King              6.272147
Roadhouse Restaurant     5.953225
KFC                      5.421687
Chef Express             3.508150
Billy Tacos              2.445074
Starbucks                1.700921
Spontini                 0.885897
Popeyes                  0.531538
Spizzico                 0.531538
Five Guys                0.354359
Burgez                   0.354359
Subway                   0.318923
Name: proportion, dtype: float64

From the above two results, we can see that the dataset is not evenly distributed across brand. Some brands have many outlets, while others have only a small number of observations. This matters because overall results may be influenced more by the largest brands, and comparisons for smaller brands may be less reliable. Graph is present in the summary_visualization file

In [15]:
outlets["Menu_Category"].value_counts()

Menu_Category
Burgers          1230
Piadine           528
Bar/Café          381
Pizza             221
Fried Chicken     168
Steakhouse        168
Mexican            69
Coffee             48
Sandwiches          9
Name: count, dtype: int64

In [16]:
outlets["Menu_Category"].value_counts(normalize= True) * 100

Menu_Category
Burgers          43.586109
Piadine          18.710135
Bar/Café         13.501063
Pizza             7.831325
Fried Chicken     5.953225
Steakhouse        5.953225
Mexican           2.445074
Coffee            1.700921
Sandwiches        0.318923
Name: proportion, dtype: float64

In [17]:
# Count outlets by region

outlets["Region"].value_counts()


Region
Lombardia                800
Lazio                    415
Piemonte                 253
Veneto                   246
Emilia-Romagna           245
Toscana                  178
Campania                 125
Sicilia                  104
Liguria                   64
Marche                    63
Puglia                    62
Abruzzo                   56
Sardegna                  54
Friuli-Venezia Giulia     50
Calabria                  34
Trentino-Alto Adige       31
Umbria                    30
Valle d'Aosta              5
Basilicata                 4
Molise                     3
Name: count, dtype: int64

In [18]:
# Percentage share of outlets by region

outlets["Region"].value_counts(normalize=True) * 100


Region
Lombardia                28.348689
Lazio                    14.705882
Piemonte                  8.965273
Veneto                    8.717222
Emilia-Romagna            8.681786
Toscana                   6.307583
Campania                  4.429483
Sicilia                   3.685330
Liguria                   2.267895
Marche                    2.232459
Puglia                    2.197023
Abruzzo                   1.984408
Sardegna                  1.913536
Friuli-Venezia Giulia     1.771793
Calabria                  1.204819
Trentino-Alto Adige       1.098512
Umbria                    1.063076
Valle d'Aosta             0.177179
Basilicata                0.141743
Molise                    0.106308
Name: proportion, dtype: float64

From the above two results, we can see that outlets are concentrated in certain regions. Lombardia has the largest share of QSR outlets, while smaller regions such as Molise and Valle d'Aosta have very few outlets. This matters because regional concentration may be related to population size, tourism, income, and density.


In [19]:
# Count outlets by location type

outlets["Location_Type"].value_counts()


Location_Type
Commercial Area     1001
Shopping Mall        910
Service Area         368
City Center          250
Transport Hub        226
Residential Area      67
Name: count, dtype: int64

In [20]:
# Percentage share of outlets by location type

outlets["Location_Type"].value_counts(normalize=True) * 100


Location_Type
Commercial Area     35.471297
Shopping Mall       32.246634
Service Area        13.040397
City Center          8.858965
Transport Hub        8.008505
Residential Area     2.374203
Name: proportion, dtype: float64

These show which types of locations are most common for QSR outlets. Commercial areas and shopping malls dominate, suggesting that QSR outlets are mainly located in high-traffic consumer spaces.


In [21]:
# Count outlets by cleaned price range

outlets["Price_Range_Cleaned"].value_counts()


Price_Range_Cleaned
EUR 10-20    1328
EUR 1-10     1281
EUR 20-30     213
Name: count, dtype: int64

In [22]:
# Percentage share of outlets by cleaned price range

outlets["Price_Range_Cleaned"].value_counts(normalize=True) * 100


Price_Range_Cleaned
EUR 10-20    47.058824
EUR 1-10     45.393338
EUR 20-30     7.547838
Name: proportion, dtype: float64

The results with the price range show that most outlets are in the lower and middle price ranges. That tells us most of the outlets are affordable QSRs.

In [23]:
# Average customer rating and review count by brand

outlets.groupby("Brand")[["Customer_Rating", "Review_Count"]].mean().sort_values(
    by="Customer_Rating",
    ascending=False
)


,Customer_Rating,Review_Count
Brand,,
Popeyes,4.220000,417.866667
La Piadineria,4.089773,293.956439
Old Wild West,4.063786,1928.279835
Alice Pizza,4.063536,407.497238
KFC,4.049673,1171.241830
Spizzico,3.986667,4017.333333
Burger King,3.860452,2331.966102
Burgez,3.850000,441.400000
Roadhouse Restaurant,3.849405,1823.029762


This table shows the average customer rating and average review count for each QSR brand. It helps compare how brands perform in terms of customer satisfaction and customer attention.

However, these averages should be interpreted carefully because the number of outlets is not the same for every brand. Brands with many outlets, such as McDonald's or La Piadineria, have more stable averages because they are based on many observations. Brands with only a few outlets can have averages that are strongly influenced by one or two locations.

Therefore, this table is useful for identifying initial patterns, but it shoul d not be treated as a final ranking of brand performance without also considering the number of outlets for each brand.


In [24]:
# Brand-level summary with outlet count, average rating, and average review count

outlets.groupby("Brand").agg(
    Outlet_Count=("Outlet ID", "count"),
    Average_Rating=("Customer_Rating", "mean"),
    Average_Review_Count=("Review_Count", "mean")
).sort_values(by="Average_Rating", ascending=False)


,Outlet_Count,Average_Rating,Average_Review_Count
Brand,,,
Popeyes,15,4.220000,417.866667
La Piadineria,528,4.089773,293.956439
Old Wild West,243,4.063786,1928.279835
Alice Pizza,181,4.063536,407.497238
KFC,153,4.049673,1171.241830
Spizzico,15,3.986667,4017.333333
Burger King,177,3.860452,2331.966102
Burgez,10,3.850000,441.400000
Roadhouse Restaurant,168,3.849405,1823.029762


This table adds the number of outlets for each brand alongside the average rating and average review count. This makes the comparison more reliable because we can see whether each brand average is based on many outlets or only a few. Brands with larger outlet counts provide more stable averages, while brands with small outlet counts should be interpreted carefully.


In [25]:
# Region-level summary with outlet count, average rating, and average review count

outlets.groupby("Region").agg(
    Outlet_Count=("Outlet ID", "count"),
    Average_Rating=("Customer_Rating", "mean"),
    Average_Review_Count=("Review_Count", "mean")
).sort_values(by="Outlet_Count", ascending=False)


,Outlet_Count,Average_Rating,Average_Review_Count
Region,,,
Lombardia,800,3.829500,1624.176250
Lazio,415,4.004337,1803.279518
Piemonte,253,3.822925,1672.359684
Veneto,246,3.807724,1764.500000
Emilia-Romagna,245,3.807347,3142.967347
Toscana,178,3.846629,1971.803371
Campania,125,3.810400,1817.952000
Sicilia,104,3.895192,1473.086538
Liguria,64,3.850000,2551.312500


This table summarizes QSR outlets by region. It shows where outlets are concentrated and compares average customer rating and average review count across regions.

Regions with many outlets, such as Lombardia and Lazio, have more stable averages because they are based on many observations. Regions with very few outlets, such as Molise or Valle d'Aosta, should be interpreted more carefully because a small number of outlets can strongly affect the average rating or review count.

In [26]:
outlets.loc[outlets['Customer_Rating'] == 0]

,Outlet ID,Brand,Region,Province,Province_Shorthand,City,Menu_Category,Price_Range,Customer_Rating,Review_Count,Latitude,Longitude,Location_Type,Address,Source,Price_Range_Cleaned,Price_Range_Imputed
